### ISSUE TITLE
Glitch while updating a pano

### ISSUE THREAD

**Issue by KhaledGabr** [facebook/react-360#275]
I have two panorama:
The first one has a rotateY of 0.
The second one has rotateY of 90

while updating using setState [ for millisecond while the second image loads ] the first image gets rotated 90.

Expected:
The first image should not get rotated 90 degree while the second image loads.

This is the code of the component
```
            <Pano
                source = {asset( this.props.sceneNum + '.jpg')}
                
                style={{
                    transform: [
                        {rotateY : this.props.orientation},
                    ]
                }}
                />
```

**Comment by amberroy** [comment]
Keep in mind that if you want to change values using `setState` you'll need to access `this.state` and not `this.props` see the [TourSample](https://github.com/facebook/react-vr/blob/master/Examples/TourSample/index.vr.js) for an example of setting rotation and source via `setState`.

**Comment by KhaledGabr** [comment]
Yes I used this.state, but in order to pass the values to the inner component I used this.props.


`            <Scene sceneNum={this.state.currentScene} orientation={ orientationData[this.state.currentScene] } >`

**Comment by mikearmstrong001** [comment]
> while updating using setState [ for millisecond while the second image loads ] the first image gets rotated 90.

You will need to manage this with an onLoad callback on the Pano. Images are always async loaded while the rotate change happens immediately. 
In circumstances like this I'd recommend either
- create two Panos, one hidden with the new setup and a key specified to be related to the url of the source. When new one completes load no longer create the old one. The keys are used to help react VR internals know which one has been removed
eg 
```
<Pano key={this.props.sceneNum+'.jpg} source = {asset( this.props.sceneNum + '.jpg')} ... />
<Pano key={this.props.newSceneNum+'.jpg} source = {asset( this.props.newSceneNum + '.jpg')} ... />
```
- fade the old pano to black and then apply new pano image, make the rotation and fade in when ready.

**Comment by andelar** [comment]
It would be nicer to fade between the two panos instead of fade to black and then back. On a bright sunny day moving from position A to position B, that becomes an ugly transition. Need opacity on Pano, see #264 

Another issue with the suggested approach is that onLoadEnd is called when the pano is loaded, which seems to be before it is rotated, so hard to know when to show it, unless delaying all transitions some arbitrary time. If you look at the Tour example you have recently built, you have the same issue. Each new view flickers in with 0 degree rotation, and then rotates some milliseconds later. I'll file a separate item for this.

**Comment by andelar** [comment]
For clarity, remaining question is opacity for a nice transition.


The original rotation issue raised by KhaledGabr can be fixed as you pointed out Mike by

        <Pano key="urlofthispano" style={{display:'flex'}} source={this.state.source} />
        <Pano key="urlofthatpano" style={{display:'none'}} source={{uri: PISA_IMGS}} onLoadEnd={() => console.log('loaded')}/>

When the second pano loaded, I also waited a few frames before switching the views for any transform rotateY on pano2 to finish.

(Note I changed to onLoadEnd, or it didn't work well on cubemaps in 1.4.0.) 
Thank you!
